dashboard/app/models/concerns/serialized_properties.rb

```diff
@@ -29,6 +29,10 @@ def init_internals
     super
   end
 
+  def property_before_save(key)
+    properties_previous_change&.any? ? properties_previous_change.dig(0, key) : properties[key]
+  end
```

Comment:
This is to support knowing what the prior value was to set `state_before` in the user event and to encapsulate that wonky logic to handle `nil`.